src/xmipp/libraries/data/unitCell.cpp

```diff
@@ -305,85 +305,101 @@ void UnitCell::tetrahedralSymmetry(const Matrix1D<double> & _centroid,
 		const Matrix1D<double> & _3fpp, const double expanded) {
 	//amplified_expanded is the resulting product of multiplying the expanded selected value by the factor 25, which could be changed
 	double amplified_expanded = expanded * 25;
+	
+	Matrix1D<double> _2f = (_3f + _3fp) / 2.;
+	Matrix1D<double> _2fp = (_3f + _3fpp) / 2.;
+	Matrix1D<double> _f = (_3f + _3fp + _3fpp) / 3.;
+
+	//vectors that join a symmetry axis with the next only.
+	Matrix1D<double> _f_to_2f = _2f - _f;
+	Matrix1D<double> _2f_to_3f = _3f - _2f;
+	Matrix1D<double> _3f_to_2fp = _2fp - _3f;
+	Matrix1D<double> _2fp_to_f = _f - _2fp;
+
+	// vector perpendicular to the triangle face.
+	Matrix1D<double> planeVector = vectorProduct(_2f_to_3f, _3f_to_2fp);
+	planeVector.selfNormalize();
 
 	//vectExpansion: vectors perpendicular to the unit cell edges
-	Matrix1D<double> vectorPlane_0 = vectorProduct(_3f, _3fp);
+	Matrix1D<double> vectorPlane_0 = vectorProduct(_f_to_2f, planeVector);
```

Comment:
names starting with underscore are typically reserved and should not be used

Reply:
We have modified the names of almost everything (except _minZ and _maxZ because the recommended names start with underscore)